src/platform/backends/lxd/lxd_request.cpp

```diff
@@ -82,29 +80,39 @@ const QJsonObject lxd_request_common(const std::string& method, QUrl& url, int t
     if (reply->error() == QNetworkReply::ContentNotFoundError)
         throw mp::LXDNotFoundException();
 
-    if (reply->error() != QNetworkReply::NoError)
-        throw std::runtime_error(fmt::format("{}: {}", url.toString(), reply->errorString()));
+    if (reply->error() == QNetworkReply::OperationCanceledError)
+        throw mp::LXDRuntimeError(
+            fmt::format("Timeout getting response for {} operation on {}", method, url.toString()));
 
     auto bytearray_reply = reply->readAll();
     reply->deleteLater();
 
+    if (bytearray_reply.isEmpty())
+        throw mp::LXDRuntimeError(fmt::format("Empty reply received for {} operation on {}", method, url.toString()));
+
     QJsonParseError json_error;
     auto json_reply = QJsonDocument::fromJson(bytearray_reply, &json_error);
 
     if (json_error.error != QJsonParseError::NoError)
-        throw std::runtime_error(fmt::format("{}: {}", url.toString(), json_error.errorString()));
+        throw mp::LXDRuntimeError(fmt::format("Error parsing JSON response for {}: {}\n{}", url.toString(),
+                                              json_error.errorString(), bytearray_reply));
 
     if (json_reply.isNull() || !json_reply.isObject())
-        throw std::runtime_error(fmt::format("Invalid LXD response for url {}: {}", url.toString(), bytearray_reply));
+        throw mp::LXDRuntimeError(fmt::format("Invalid LXD response for {}: {}", url.toString(), bytearray_reply));
 
     mpl::log(mpl::Level::trace, request_category, fmt::format("Got reply: {}", QJsonDocument(json_reply).toJson()));
```

Comment:
We won't see this if JSON fails to parse. Should we move this up (and skip the parse + dump loop)?

Reply:
Um, well, if the JSON fails to parse, we won't get anything from `QJsonDocument(json_reply).toJson()` anyways, so I don't see why we need to move this up....unless I misunderstand what you mean here.

Reply:
I mean that when we fail to parse the JSON further up, we won't log the contents of the HTTP reply. Which, when it fails to parse, will likely be interesting.

Reply:
If it fails to parse the JSON further up, then `QJsonDocument(json_reply).toJson()` here isn't going to work, thus is won't log what we expect.